src/componententity.cpp

```diff
@@ -192,6 +201,64 @@ void ComponentEntity::removeAllUnits()
     mPimpl->mUnits.clear();
 }
 
+bool ComponentEntity::hasUnits(const std::string &name) const
+{
+    auto result = mPimpl->findUnits(name);
+    return result != mPimpl->mUnits.end();
+}
+
+UnitsPtr ComponentEntity::getUnits(size_t index)
+{
+    return mPimpl->mUnits.at(index);
+}
+
+const UnitsPtr& ComponentEntity::getUnits(size_t index) const
+{
+    return mPimpl->mUnits.at(index);
+}
+
+UnitsPtr ComponentEntity::getUnits(const std::string &name)
+{
+    auto result = mPimpl->findUnits(name);
+    size_t index = result - mPimpl->mUnits.begin();
+    return mPimpl->mUnits.at(index);
```

Comment:
It's clearly not (that) important, but why don't we simply have the following? (Same elsewhere in the code.)

```
return mPimpl->mUnits.at(mPimpl->findUnits(name) - mPimpl->mUnits.begin());
```